### PR TITLE
minor: Improve testing of math scalar functions

### PR DESCRIPTION
**Pull request by andygrove**
## Which issue does this PR close?

N/A

## Rationale for this change

There were multiple issues with the `various math scalar functions` test:

- It was testing multiple expressions in one query, including `abs`, which is not supported, so the entire query fell back to Spark and did not test Comet
- The input data did not include boundary values
- The test ran with and without dictionary encoding enabled but did not have any duplicate values so was not actually using dictionary encoding
- There were no assertions that anything actually ran in Comet

I discovered a new bug during this work - https://github.com/apache/datafusion-comet/issues/1897

## What changes are included in this PR?

Rewrite the test to run one expression at a time and ensure that the query ran in Comet

### CODE REVIEW

File: spark/src/test/scala/org/apache/comet/CometExpressionSuite.scala [hunk 1]
```diff
@@ -1217,38 +1217,54 @@ class CometExpressionSuite extends CometTestBase with AdaptiveSparkPlanHelper {
...
-      0,
-      +1,
-      Double.MinValue,
-      Double.MaxValue,
-      Double.NaN,
-      Double.MinPositiveValue,
-      Double.PositiveInfinity,
-      Double.NegativeInfinity)
-    val testValuesRepeated = testValues.flatMap(v => Seq.fill(1000)(v))
+    val testValuesRepeated = doubleValues.flatMap(v => Seq.fill(1000)(v))
```
Comment (mbutrovich) [inline spark/src/test/scala/org/apache/comet/CometExpressionSuite.scala:1267]: Is this testing that Comet expressions work with dictionary encoded values?

Comment (andygrove) [reply spark/src/test/scala/org/apache/comet/CometExpressionSuite.scala:1267]: That is the intent, although there are no assertions to ensure that is what is actually happening
